v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/GcpToNeo4j.java

```diff
@@ -0,0 +1,339 @@
+import com.google.cloud.teleport.v2.neo4j.model.enums.ActionExecuteAfter;
+import com.google.cloud.teleport.v2.neo4j.model.enums.ArtifactType;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.JobSpecMapper;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.OptionsParamsMapper;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.SourceQuerySpec;
+import com.google.cloud.teleport.v2.neo4j.model.helpers.TargetQuerySpec;
+import com.google.cloud.teleport.v2.neo4j.model.job.Action;
+import com.google.cloud.teleport.v2.neo4j.model.job.ActionContext;
+import com.google.cloud.teleport.v2.neo4j.model.job.JobSpec;
+import com.google.cloud.teleport.v2.neo4j.model.job.OptionsParams;
+import com.google.cloud.teleport.v2.neo4j.model.job.Source;
+import com.google.cloud.teleport.v2.neo4j.model.job.Target;
+import com.google.cloud.teleport.v2.neo4j.providers.IProvider;
+import com.google.cloud.teleport.v2.neo4j.providers.ProviderFactory;
+import com.google.cloud.teleport.v2.neo4j.transforms.GcsLogTransform;
+import com.google.cloud.teleport.v2.neo4j.transforms.Neo4jRowWriterTransform;
+import com.google.cloud.teleport.v2.neo4j.utils.BeamBlock;
+import com.google.cloud.teleport.v2.neo4j.utils.ModelUtils;
+import com.google.cloud.teleport.v2.neo4j.utils.ProcessingCoder;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.util.List;
+import org.apache.beam.repackaged.core.org.apache.commons.lang3.StringUtils;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.FileSystems;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.Wait;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.Row;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Dataflow template which reads BigQuery data and writes it to Neo4j. The source data can be
+ * either a BigQuery table or an SQL query.
+ */
+public class GcpToNeo4j {
+
+    private static final Logger LOG = LoggerFactory.getLogger(GcpToNeo4j.class);
+
+    private static final Gson gson = new GsonBuilder().setPrettyPrinting().create();
+
+    public OptionsParams optionsParams;
+    ConnectionParams neo4jConnection;
+    JobSpec jobSpec;
+    Pipeline pipeline;
+
+    /**
+     * Main class for template.  Initializes job using run-time on pipelineOptions.
+     *
+     * @pipelineOptions framework supplied arguments
+     */
+    public GcpToNeo4j(final Neo4jFlexTemplateOptions pipelineOptions) {
+        ////////////////////////////
+        // We need to initialize pipeline in order to create context for Gs and Bq file system
+        final String jobName = pipelineOptions.getJobName() + "-" + System.currentTimeMillis();
```

Comment:
What's the reason for this job name override?
The default job name already contains date/time in it.

Reply:
When using DirectRunner the jobName does not get a date timestamp, AFAIK.